Model/Behavior/SoftDeleteBehavior.php

```diff
@@ -346,7 +346,7 @@ protected function _softDeleteAssociations(Model $model, $active) {
 
 					$conditions = $model->{$parentModel}->{$assocType}[$assoc]['conditions'];
 					if (!is_array($conditions)) {
-						$model->{$parentModel}->{$assocType}[$assoc]['conditions'] = [];
+						$conditions = [];
```

Comment:
Sounds good, but does this need to be written back to the model? Looks like the code after does only modify a local variable.